libraries/chain/controller.cpp

```diff
@@ -763,12 +763,12 @@ struct building_block {
                   // Create the valid structure for validating_bsp if it does not
                   // have one.
                   if (!validating_bsp->valid) {
-                     validating_bsp->valid = bb.parent.new_valid(bhs, action_mroot);
+                     validating_bsp->valid = bb.parent.new_valid(bhs, action_mroot, bb.parent.strong_digest);
```

Comment:
`bb.parent.strong_digest` does not seem correct. It should be the strong digest of the next block state. Here we don't have strong digest yet. Have to compute it by `bhs.compute_finality_digest()`.

Reply:
Done